ArduPlane/tiltrotor.cpp

```diff
@@ -345,9 +294,16 @@ void QuadPlane::tilt_compensate(float *thrust, uint8_t num_motors)
     }
     if (in_vtol_mode()) {
         // we are transitioning to VTOL flight
-        tilt_compensate_up(thrust, num_motors);
+        const float tilt_factor = cosf(radians(tilt.current_tilt*90));
+        tilt_compensate_angle(thrust, num_motors, tilt_factor, 1);
     } else {
-        tilt_compensate_down(thrust, num_motors);
+        float inv_tilt_factor;
+        if (tilt.current_tilt > 0.98f) {
+            inv_tilt_factor = 1.0 / cosf(radians(0.98f*90));
+        } else {
+            inv_tilt_factor = 1.0 / cosf(radians(tilt.current_tilt*90));
+        }
```

Comment:
```suggestion
        float inv_tilt_factor = 1.0 / cosf(radians(MAX(tilt.current_tilt,0.98f)*90));
   
```

Should be equivalent, I guess the goal is to avoid a divide by 0

Reply:
Also, the max value of this is 31.8, seems quite high....